server/jamovi/server/instancemodel.py

```diff
@@ -49,19 +52,60 @@ def get_column_by_id(self, id):
         else:
             raise KeyError('No such column: ' + str(id))
 
-    def append_column(self, name, import_name=None, id=-1):
+    def get_transform_by_id(self, id):
+        for transform in self.transforms:
+            if transform.id == id:
+                return transform
+        else:
+            raise KeyError('No such transform: ' + str(id))
+
+    def remove_transform(self, id):
+        i = 0
+        while i <  len(self._transforms):
+            transform = self._transforms[i]
+            if transform.id == id:
+                del self._transforms[i]
+            else:
+                i += 1
+
+
+    def append_transform(self, name, id=0):
+        use_id = self._transform_next_id
+        if id != 0:
+            if id < self._transform_next_id:
+                for existing_transform in self.transforms:
+                    if existing_transform.id == id:
+                        raise KeyError('Transform id already exists: ' + str(id))
+            elif id > self._transform_next_id:
+                self._transform_next_id = id
+            use_id = id
+
+        new_transform = Transform()
+        new_transform.id = use_id
+        new_transform.name = name
+        if use_id == self._transform_next_id:
+            self._transform_next_id += 1
+        self._transforms.append(new_transform)
+
+        return new_transform;
 
+    def append_column(self, name, import_name=None, id=-1):
```

Comment:
we can get rid of the `id=-1` here can't we? i appreciate this hasn't changed in this commit, but it's bugged me for a while!